src/ASM/LR/ASMu3Drecovery.C

```diff
@@ -157,25 +157,26 @@ bool ASMu3D::assembleL2matrices (SparseMatrix& A, StdVector& B,
       Go::BasisDerivs spl2;
       int ielp = group[t][e];
       const LR::Element* elm = proj->getElement(ielp);
-      int iel = lrspline->getElementContaining(elm->midpoint())+1;
+      int iel = lrspline->getElementContaining(elm->midpoint()) + 1;
+      int ielG = geo->getElementContaining(elm->midpoint()) + 1;
 
       if (continuous)
       {
         // Set up control point (nodal) coordinates for current element
         if (!this->getElementCoordinates(Xnod,iel)) {
           ok = false;
           continue;
-        } else if ((dV = 0.125*this->getParametricVolume(iel)) < 0.0) {
+        } else if ((dV = 0.125*elm->volume()) < 0.0) {
           ok = false; // topology error (probably logic error)
```

Comment:
...but then this comment should  vanish as well. Think that also is inherited from the struct case where it was associated with the possible zero knot-spans, which we don't have in the LR-case.

Reply:
this can happen if we have a lhs geometry, which can be said to be a topology error, but i have removed it.